packages/fuels/src/cli/commands/dev/autoStartFuelCore.ts

```diff
@@ -1,15 +1,86 @@
-import { defaultConsensusKey } from '@fuel-ts/utils';
+import { defaultChainConfig, defaultConsensusKey } from '@fuel-ts/utils';
+import { launchNode } from '@fuel-ts/wallet/test-utils';
+import type { ChildProcessWithoutNullStreams } from 'child_process';
+import { mkdirSync, writeFileSync } from 'fs';
+import { dirname, join } from 'path';
+import { getPortPromise } from 'portfinder';
 
 import type { FuelsConfig } from '../../types';
+import { getBinarySource } from '../../utils/getBinarySource';
+import { log, loggingConfig } from '../../utils/logger';
 
-import type { FuelCoreNode } from './startFuelCore';
-import { startFuelCore } from './startFuelCore';
+export type FuelCoreNode = {
+  bindIp: string;
+  accessIp: string;
+  port: number;
+  providerUrl: string;
+  chainConfig: string;
+  killChildProcess: () => void;
+};
+
+export type KillNodeParams = {
+  core: ChildProcessWithoutNullStreams;
+  killFn: (pid: number) => void;
+  state: {
+    isDead: boolean;
+  };
+};
+
+export const killNode = (params: KillNodeParams) => () => {
+  const { core, state, killFn } = params;
+  if (core.pid && !state.isDead) {
+    state.isDead = true;
+    killFn(Number(core.pid));
+  }
+};
```

Comment:
We no longer need this method since you already do it inside the `launchNode`, no?

https://github.com/FuelLabs/fuels-ts/blob/2154aae9bf8149df3df05e2e9379ffd025f39ec8/packages/wallet/src/test-utils/launchNode.ts#L64